### PR TITLE
[MRG] Add tests for port mapping conversion

### PR DESCRIPTION
**Pull request by betatim**
Wanted to quickly add some tests for the port mapping function and then ended up rewriting it.

I tried to follow https://docs.docker.com/config/containers/container-networking/#published-ports because I've never used a port mapping more complex than `8888:8888`.

### CODE REVIEW

**Comment by choldgraf** [comment]
this seems reasonable to me, and I am always in favor of doing things w/o regex for maintainability reasons...though this also means it's a bit hard for me to know if the new way of doing things is really doing the same as the old regex stuff. That said, tests are passing so I will leave a day or so for this one to marinate, and will plan on merging if nobody protests

**Comment by betatim** [comment]
I don't know why there was a regex either and I can't read regex. The docs were out of date/didn't describe what the function actually did either :-/ I think overall less regex is better.

File: repo2docker/utils.py [hunk 1]
```diff
@@ -68,24 +68,17 @@ def chdir(path):
         os.chdir(old_dir)
 
 
-def validate_and_generate_port_mapping(port_mapping):
+def validate_and_generate_port_mapping(port_mappings):
     """
-    Validate the port mapping list and return a list of validated tuples.
-
-    Each entry in the passed port mapping list will be converted to a
```
Comment (yuvipanda) [inline repo2docker/utils.py]: Is there a reason you deleted this docstring?

File: repo2docker/utils.py [hunk 2]
```diff
@@ -94,55 +87,50 @@ def validate_and_generate_port_mapping(port_mapping):
...
-        Valid port mappings are:
-        - `127.0.0.1:90:900`
-        - `:999`  (match to any host port)
-        - `999:999/tcp` (bind 999 host port to 999 tcp container port)
-
-        Invalid port mapping:
-        - `127.0.0.1::999` (even though docker accepts it)
-        - other invalid ip address combinations
     """
-    reg_regex = re.compile(r"""
```
Comment (yuvipanda) [inline repo2docker/utils.py]: less regex is good!

File: repo2docker/utils.py (hunk 1 above)
Comment (betatim) [reply repo2docker/utils.py]: I tried to update it as parts weren't describing what actually happened anymore.

Not sure what the best way to describe the returned dictionary is, update the docstring to link to the docker-py method that consumes this port mapping.

**Comment by yuvipanda** [comment]
Thanks, @betatim

**Pull request merged**